src/frontend/js/components/SearchSuggestField/index.tsx

```diff
@@ -87,7 +87,10 @@ const SearchSuggestField = ({ context }: CommonDataProps) => {
       });
     }
   };
-  const updateCourseSearchParamsDebounced = debounce(searchAsTheUserTypes, 500, { maxWait: 1100 });
+  const updateCourseSearchParamsDebounced = useMemo(
+    () => debounce(searchAsTheUserTypes, 500, { maxWait: 1100 }),
+    [],
+  );
```

Comment:
I don't understand how it is going to work. If you pass [] as params then they are never going to change and memoization is forever so debounce is called only once?

Reply:
Yes, the debounce method is called once, so only once timer is instantiated during the component life cycle. Previously, we called debounce at every render, so a timer was created each time and debounce cannot work.

Reply:
Would be nice if this can be tested ^^

Reply:
Yes I'm currently fighting to fix tests and remove this TODO ^^.
https://github.com/openfun/richie/blob/f2c68abe06d2e711f5bea0c9835f84de82777a81/src/frontend/js/components/SearchSuggestField/index.spec.tsx#L470